CRM/Contribute/Form/Contribution/Confirm.php

```diff
@@ -2435,12 +2435,12 @@ protected function completeTransaction($result, $contributionID) {
    *
    * @param string $message
    */
-  protected function bounceOnError($message) {
+  protected function bounceOnError($message): void {
     CRM_Core_Session::singleton()
-      ->setStatus(ts("Payment Processor Error message :") .
+      ->setStatus(ts('Payment Processor Error message :') .
         $message);
     CRM_Utils_System::redirect(CRM_Utils_System::url('civicrm/contribute/transact',
-      "_qf_Main_display=true&qfKey={$this->_params['qfKey']}"
+      '_qf_Main_display=true&qfKey=' . ($this->_params['qfKey'] ?? NULL)
```

Comment:
this prevents notices at this point from masking the error

Reply:
Should it be `?? ''` ?

Lint style issue also below.

Reply:
@demeritcowboy I don't think that matters does it - is just doesn't add anything on? I have done that elsewhere & haven't see it upset php - but maybe it does at some version level?

Reply:
Maybe just a philosophical question then: What does it mean to concatenate oneself with null?